### PR TITLE
CI: add `prettier` checks

### PR DESCRIPTION
**Pull request by drupol**
Hello,

In this PR, I am proposing the integration of [`Prettier`](https://prettier.io/) in this project, aimed at enhancing the consistency of our `.md` and `.json` files across the entire codebase. Prettier, being an opinionated code formatter, will help us maintain a uniform code style and improve code readability.

One of the significant advantages of Prettier is its ability to read the existing `.editorconfig` and format files accordingly, ensuring that we adhere to established coding guidelines while also automating the formatting process.

Let me know what you think on this!

### CODE REVIEW

File: UPGRADE-2.0.md [hunk 1]
```diff
@@ -2,76 +2,76 @@
...
 
 ### Detailed differences in event flow during dependency resolution, composer updates and installs
 
 #### Composer v1
 
-- Composer resolves dependencies (dispatching PRE/POST_DEPENDENCIES_SOLVING)
-- It then iterates over all packages one by one (dispatching PRE_PACKAGE_INSTALL/UPDATE/UNINSTALL, then PRE_FILE_DOWNLOAD if needed, then POST_PACKAGE_\*)
-- And finally writes the lock file at the end
+-   Composer resolves dependencies (dispatching PRE/POST_DEPENDENCIES_SOLVING)
+-   It then iterates over all packages one by one (dispatching PRE*PACKAGE_INSTALL/UPDATE/UNINSTALL, then PRE_FILE_DOWNLOAD if needed, then POST_PACKAGE*\*)
```
Comment (fredden) [inline UPGRADE-2.0.md:58]: This looks like the auto-fixer broken the intended formatting. Perhaps we need to quote `PRE_PACKAGE_INSTALL` with ticks?
```suggestion
-   It then iterates over all packages one by one (dispatching `PRE_PACKAGE_INSTALL`/`UPDATE`/`UNINSTALL`, then `PRE_FILE_DOWNLOAD` if needed, then `POST_PACKAGE_*`)
```

File: UPGRADE-2.0.md [hunk 2]
```diff
@@ -2,76 +2,76 @@
...
+-   It then writes the lock file and that's the end of the update
 
 Install then does:
 
-- Dispatches PRE_OPERATIONS_EXEC with the full list of operations to be executed
-- Downloads all the packages not in cache yet in parallel (dispatching PRE_FILE_DOWNLOAD for those not in cache yet)
-- It then iterates over all packages and executes updates/installs/uninstalls in parallel (dispatching PRE_PACKAGE_INSTALL/UPDATE/UNINSTALL then POST_PACKAGE_\* but one package started last may finish installing before another is done for example).
+-   Dispatches PRE_OPERATIONS_EXEC with the full list of operations to be executed
+-   Downloads all the packages not in cache yet in parallel (dispatching PRE_FILE_DOWNLOAD for those not in cache yet)
+-   It then iterates over all packages and executes updates/installs/uninstalls in parallel (dispatching PRE*PACKAGE_INSTALL/UPDATE/UNINSTALL then POST_PACKAGE*\* but one package started last may finish installing before another is done for example).
```
Comment (fredden) [inline UPGRADE-2.0.md:74]: This looks like the auto-fixer broken the intended formatting. Perhaps we need to quote `PRE_PACKAGE_INSTALL` with ticks?
```suggestion
-   It then iterates over all packages and executes updates/installs/uninstalls in parallel (dispatching `PRE_PACKAGE_INSTALL`/`UPDATE`/`UNINSTALL` then `POST_PACKAGE_*` but one package started last may finish installing before another is done for example).
```

File: UPGRADE-2.0.md (hunk 1 above)
Comment (drupol) [reply UPGRADE-2.0.md]: Thanks, I'm now including a intermediary commit fixing those things.

File: UPGRADE-2.0.md (hunk 2 above)
Comment (drupol) [reply UPGRADE-2.0.md]: Thanks, I'm now including a intermediary commit fixing those things.

**Comment by Seldaek** [comment]
Thanks but no thanks.. I don't see the value add here, it's mostly "fixing" a few quotes and then reindenting all bullet lists with 3 spaces instead of one, which I don't see as an improvement, as I always write them with one it'll just nag me forever.

**Comment by Seldaek** [comment]
Sorry i have an allergy to most linting tools, it's so rigid and tends to just get in the way when working with open source IMO. It probably makes more sense in larger teams where many people work for longer periods on the same stuff. Anyway..

**Comment by drupol** [comment]
I totally understand regarding linting tools. They can indeed feel a bit too restrictive at times, especially when working in open source projects where there is often a wide range of coding styles and philosophies. It's an absolutely normal perspective and I'm not offended at all.

To me, PRs are by their very nature, a proposal, if you will. They provide an opportunity to suggest changes, but they're by no means a mandate. As such, I fully respect and appreciate your ability to say yes or no as a maintainer. That's part of what makes the open source community so dynamic and effective.

Rest assured, this won't prevent me from contributing in the future. Quite the opposite, in fact – I look forward to continuing to engage with and contribute to this project in whatever ways I can.

Cheers ! :beer:

**Pull request closed without merging**